docs/compile.md

```diff
@@ -188,7 +188,69 @@ It's up to you to decide what the output is.
 
 ### kadet
 
-This input type is experimental. See <https://github.com/deepmind/kapitan/pull/190> for its usage.
+Kadet is an extensible input type that enables you to generate templates using python. The key benefit being the ability to utilize familiar programing principles while having access to kapitan's powerful inventory system. 
+
+A library that defines resources as classes using the Base Object class (BaseObj) is required. These can then be utilized within components to render output. 
+
+The following functions are provided by the class BaseObj.
+
+Method definitions:<br/>
+
+* new: Provides parameter checking capabilities<br/>
+* body: Enables in-depth parameter configuration<br/>
+
+Method functions:<br/>
+
+* root: Defines values that will be compiled into the output<br/>
+* need: Ability to check & define input parameters<br/>
+* update_root: Updates the template file associated with the class<br/>
+
+A class can be a resource such as a kubernetes Deployment as shown here:
+
+```
+class Deployment(BaseObj):
+    def new(self):
+        self.need("name", "name string needed")
+        self.need("labels", "labels dict needed")
+        self.need("containers", "containers dict needed")
+        self.update_root("lib/kubelib/deployment.yml")
+
+    def body(self):
+        self.root.metadata.name = self.kwargs.name
+        self.root.metadata.namespace = inv.parameters.target_name
+        self.root.spec.template.metadata.labels = self.kwargs.labels
+        self.root.spec.template.spec.containers = self.kwargs.containers
+```
+
+The deployment is an BaseObj which has two main funcations. New can be used to perform parameter validation & template compilation. Body is utilized to set those parameters to be rendered. `self.root.metadata.name` is a direct reference to a key in the corresponding yaml.
```

Comment:
Typo in `functions`.

Reply:
school boy error :(, should be paying more attention